### PR TITLE
[Breaking] Move global metadata, file metadata config migration to docfx

### PR DESCRIPTION
**Pull request by yufeih**
https://dev.azure.com/ceapex/Engineering/_git/Docs.Build/pullrequest/17864?_a=overview

###### Microsoft Reviewers: [Open in CodeFlow](http://wpcp.azurewebsites.net/CodeFlowProtocolProxy2.php?pullrequest=https://github.com/dotnet/docfx/pull/5260)

### CODE REVIEW

**Comment by yufeih** [comment]
Diffs are all expected `LandingData` output property reorder.

File: test/docfx.Test/lib/GlobUtilityTest.cs [hunk 1]
```diff
@@ -60,9 +60,21 @@ public class GlobUtilityTest
         [InlineData("**/[EJ]/*.{md,cs,csproj}", "Root/M/K.md, Root/J\\K.csp", false)]
         [InlineData("a**/*.md", "ab/c.md, a/b.md", true)]
         [InlineData("a**/*.md", "a/b/c.md, a/b.cs", false)]
+
+        // For backward compatibility with v2
+        [InlineData("****", "a.md, a/b.md", true)]
+        [InlineData("**.*", "a.md a/b.md", true)]
```
Comment (mingwli) [inline test/docfx.Test/lib/GlobUtilityTest.cs]: > "a.md a/b.md" [](start = 27, length = 14)

forget some commas?

File: src/docfx/build/metadata/UserMetadata.cs [hunk 2]
```diff
@@ -24,6 +25,10 @@ internal class InputMetadata
         [JsonProperty("_tocRel")]
         public string TocRel { get; set; }
 
+        [JsonProperty("_op_documentIdPathDepotMapping")]
+        public Dictionary<string, DocumentIdDepotMapping> DocumentIdDepotMapping { get; set; }
```
Comment (mingwli) [inline src/docfx/build/metadata/UserMetadata.cs]: >DocumentIdDepotMapping  [](start = 58, length = 23)

what does this server for?

Comment (yufeih) [reply src/docfx/build/metadata/UserMetadata.cs]: https://dev.azure.com/ceapex/Engineering/_git/Docs.Build?path=%2Fsrc%2FConfigMigration%2Fsrc%2Fmigration%2Fmigrator%2Fdocument-id-migrator.js&version=GBdevelop&line=22&lineEnd=22&lineStartColumn=1&lineEndColumn=122&lineStyle=plain

File: test/docfx.Test/lib/GlobUtilityTest.cs (hunk 1 above)
Comment (yufeih) [reply test/docfx.Test/lib/GlobUtilityTest.cs]: yep

File: src/docfx/build/metadata/UserMetadata.cs (hunk 2 above)
Comment (mingwli) [reply src/docfx/build/metadata/UserMetadata.cs]: thought you've already read it into GlobalMetadata?

---
In reply to: [343986084](https://github.com/dotnet/docfx/pull/5260#discussion_r343986084) [](ancestors = 343986084)

**Pull request merged**